website/src/pages/Macro/Macro.jsx

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useState, useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import RingLoader from "react-spinners/RingLoader";
+import { Swiper, SwiperSlide } from "swiper/react";
+
+import '../Macro/macro.css'
+import "swiper/css";
+import "swiper/css/pagination";
+
+import { Mousewheel, Pagination, Keyboard } from "swiper";
+import Macrobench from '../../components/Macrobench/Macrobench';
+import MacrobenchMobile from '../../components/MacrobenchMobile/MacrobenchMobile';
+
+const Macro = () => {
+    const urlParams = new URLSearchParams(window.location.search);
+    const [dropDownLeft, setDropDownLeft] = useState(urlParams.get('ltag') == null ? 'Left' : urlParams.get('ltag'));
+    const [dropDownRight, setDropDownRight] = useState(urlParams.get('rtag') == null ? 'Right' : urlParams.get('rtag'));
+    const [openDropDownLeft, setOpenDropDownLeft] = useState(58);
+    const [openDropDownRight, setOpenDropDownRight] = useState(58);
+    const [dataRefs, setDataRefs] = useState([]);
+    const [dataMacrobench, setDataMacrobench] = useState([]);
+    const [commitHashLeft, setCommitHashLeft] = useState('')
+    const [commitHashRight, setCommitHashRight] = useState('')
+    const [error, setError] = useState(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [currentSlideIndex, setCurrentSlideIndex] = useState(urlParams.get('ptag') == null ? '0' : urlParams.get('ptag'));
+    const [currentSlideIndexMobile, setCurrentSlideIndexMobile] = useState(urlParams.get('ptagM') == null ? '0' : urlParams.get('ptagM'))
+    
+    
+    useEffect(() => {
+        const fetchData = async () => {
+          try {
+            const responseRefs = await fetch(`${import.meta.env.VITE_API_URL}vitess/refs`);
+      
+            const jsonDataRefs = await responseRefs.json();
+            
+            setDataRefs(jsonDataRefs);
+            
+            setIsLoading(false)
+          } catch (error) {
+            console.log('Error while retrieving data from the API', error);
+            setError('An error occurred while retrieving data from the API. Please try again.');
```

Comment:
I see that we use `An error occurred while retrieving data from the API. Please try again.` in a lot of places. To follow DRY principles, let's extract this error string and re-use it everywhere we have hardcoded the error.